.github/workflows/ci.yml

```diff
@@ -0,0 +1,59 @@
+name: CI workflow
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.head_ref }}
+        
+    - name: Setup node
+      uses: actions/setup-node@v2
+      with:
+        node-version: 14.x
+        
+    - name: Install dependencies
+      run: npm ci
+      env:
+        CI: true
+    
+    - name: Cache npm packages
+      id: npm-cache
+      uses: actions/cache@v2
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-npm-
+        
+    - name: Lint files
+      run: npm run lint
+
+    - name: Linting failed (attempting fix...)
+      if: ${{ failure() }}
+      run: npm run lint:fix
+
+    - name: Commit fixed lint issues
```

Comment:
We don't need to have both scripts (`lint:fix` and `format`) in CI since `format` is included in `lint:fix`, but the problem is that not all lint issues can be fixed automatically, so I would leave the only `format` step in the CI that can be always applied.

Reply:
> We don't need to have both scripts (lint:fix and format) in CI since format is included in lint:fix

> so I would leave the only format step in the CI that can be always applied

What do you mean by this? One runs prettier and the other eslint so I am confused at this statement, can you clarify?

> not all lint issues can be fixed automatically

Exactly which is why the `if: ${{ failure() }}` line is there, it will only ever execute this step if a failure happened in the previous `lint` step and if it runs `lint:fix` and it can't do it for some reason then it will throw an error at that step instead. The goal of this automatic fix attempt is to reduce developer load by allowing the integration suite to automate integration into the codebase as much as possible. If something is truly unfixable automatically then the dev must fix it but otherwise we can handle it for them here.

Reply:
> What do you mean by this? One runs prettier and the other eslint so I am confused at this statement, can you clarify?

When we use ESLint and Prettier together, we need to exclude the formatting rules in ESLint that might conflict with Prettier. For that, we use `eslint-config-prettier` and `eslint-plugin-prettier` libraries that run Prettier as ESLint rules and exclude conflict rules. Since we're not running Prettier just for JS and TS files, I changed the order and run the formatting step first and then the linting JS and TS files.